boot/freeldr/freeldr/freeldr.c

```diff
@@ -74,7 +175,17 @@ VOID __cdecl BootMain(IN PCCH CmdLine)
     /* Initialize I/O subsystem */
     FsInit();
 
-    RunLoader();
```

Comment:
Looking at what you do here (as replacement), I'm under the impression that all the freeldr.ini + menu is now running inside rosload.exe, am I understanding it right?
If so, I would suggest you to keep that in the 1st-stage loader where this was meant to be, and have the rosload.exe doing only what freeldr/ntldr does. (So no menu being shown by rosload.exe; instead, needs to be shown by freeldr.sys.)

Reply:
The problem is that the menu code uses bootmgr / ntldr stuff. Moving it to rosload would fix that dependency issue, which is currently hacked.
And I don't see why freeldr would need to contain anything other than absolute minimum to load a 2nd stage loader. Why do we need rainbow colored unicorn flavoured menus in a loader that does not even show any menus, but only uses that code to print an error message on the screen?

Reply:
Like in windows, by the way, where our freeldr would be equivalent to their "bootmgr" (which shows the main menu stuff), while the new rosload would be equivalent to the winload.
Amongst other examples.